Instancio 4.1.0 fails to create protobuf message instance with google.protobuf.Struct field

**Describe the bug**
Instancio 4.1.0 fails to create protobuf message instance with google.protobuf.Struct field. Instancio 4.0.0 works fine.

<details>
<summary>Stack trace</summary>

```
Instancio encountered an error.
Please submit a bug report including the stacktrace:
https://github.com/instancio/instancio/issues

 -> Error creating instance of: class com.google.protobuf.MapField$MutabilityAwareMap

org.instancio.exception.InstancioTerminatingException: 
Instancio encountered an error.
Please submit a bug report including the stacktrace:
https://github.com/instancio/instancio/issues

 -> Error creating instance of: class com.google.protobuf.MapField$MutabilityAwareMap

	at org.instancio.internal.util.Fail.withFataInternalError(Fail.java:129)
	at org.instancio.internal.generator.util.MapGenerator.tryGenerateNonNull(MapGenerator.java:141)
	at org.instancio.internal.generator.util.MapGenerator.tryGenerateNonNull(MapGenerator.java:38)
	at org.instancio.internal.generator.AbstractGenerator.generate(AbstractGenerator.java:67)
	at org.instancio.internal.handlers.MapNodeHandler.getResult(MapNodeHandler.java:49)
	at org.instancio.internal.GeneratorFacade.getGeneratorResult(GeneratorFacade.java:131)
	at org.instancio.internal.GeneratorFacade.generateNodeValue(GeneratorFacade.java:90)
	at org.instancio.internal.InstancioEngine.generateValue(InstancioEngine.java:655)
	at org.instancio.internal.InstancioEngine.generateMap(InstancioEngine.java:197)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:159)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.createRootObjectInternal(InstancioEngine.java:105)
	at org.instancio.internal.ErrorHandler.conditionalFailOnError(ErrorHandler.java:65)
	at org.instancio.internal.InstancioEngine.createRootObject(InstancioEngine.java:100)
	at org.instancio.internal.ApiImpl.createRootObject(ApiImpl.java:210)
	at org.instancio.internal.ApiImpl.create(ApiImpl.java:177)
	at org.instancio.Instancio.create(Instancio.java:155)
	at AppTest.testCreatesInstance(AppTest.java:8)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:119)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:94)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:89)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: java.lang.NoSuchMethodException: com.google.protobuf.MapField$MutabilityAwareMap.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
	at org.instancio.internal.generator.util.MapGenerator.tryGenerateNonNull(MapGenerator.java:139)
	... 112 more


com.google.protobuf.MapField$MutabilityAwareMap.<init>()
java.lang.NoSuchMethodException: com.google.protobuf.MapField$MutabilityAwareMap.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
	at org.instancio.internal.generator.util.MapGenerator.tryGenerateNonNull(MapGenerator.java:139)
	at org.instancio.internal.generator.util.MapGenerator.tryGenerateNonNull(MapGenerator.java:38)
	at org.instancio.internal.generator.AbstractGenerator.generate(AbstractGenerator.java:67)
	at org.instancio.internal.handlers.MapNodeHandler.getResult(MapNodeHandler.java:49)
	at org.instancio.internal.GeneratorFacade.getGeneratorResult(GeneratorFacade.java:131)
	at org.instancio.internal.GeneratorFacade.generateNodeValue(GeneratorFacade.java:90)
	at org.instancio.internal.InstancioEngine.generateValue(InstancioEngine.java:655)
	at org.instancio.internal.InstancioEngine.generateMap(InstancioEngine.java:197)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:159)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.populateChildren(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.generatePojo(InstancioEngine.java:190)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:165)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:183)
	at org.instancio.internal.InstancioEngine.createRootObjectInternal(InstancioEngine.java:105)
	at org.instancio.internal.ErrorHandler.conditionalFailOnError(ErrorHandler.java:65)
	at org.instancio.internal.InstancioEngine.createRootObject(InstancioEngine.java:100)
	at org.instancio.internal.ApiImpl.createRootObject(ApiImpl.java:210)
	at org.instancio.internal.ApiImpl.create(ApiImpl.java:177)
	at org.instancio.Instancio.create(Instancio.java:155)
	at AppTest.testCreatesInstance(AppTest.java:8)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:728)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:218)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:214)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:139)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:69)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:119)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:94)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:89)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at jdk.proxy1/jdk.proxy1.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```

</details>

**Expected result**
The instance should be created correctly.

**Actual result**
Instancio raises an error.

**Link to a minimal sample project reproducing the issue or runnable sample code**
https://github.com/racevedoo/instancio-protobuf-issue, run `./gradlew test`. If the instancio version is changed to 4.0.0 in build.gradle, the problem does not occur.

- Java version: 21
- Instancio version: 4.1.0

--- a/instancio-core/src/main/java/org/instancio/internal/ErrorHandler.java
+++ b/instancio-core/src/main/java/org/instancio/internal/ErrorHandler.java
@@ -22,6 +22,7 @@ import org.instancio.internal.util.Format;
 import org.instancio.internal.util.Sonar;
 import org.instancio.internal.util.SystemProperties;
 import org.instancio.settings.Keys;
+import org.instancio.settings.Settings;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -29,7 +30,7 @@ import java.util.Optional;
 import java.util.function.Supplier;
 
 @SuppressWarnings(Sonar.CATCH_EXCEPTION_INSTEAD_OF_THROWABLE)
-final class ErrorHandler {
+public final class ErrorHandler {
     private static final Logger LOG = LoggerFactory.getLogger(ErrorHandler.class);
 
     private static final String SUPPRESSION_REASON = String.format("" +
@@ -66,7 +67,7 @@ final class ErrorHandler {
         } catch (AssertionError | InstancioTerminatingException ex) {
             throw ex;
         } catch (Throwable ex) { //NOPMD
-            if (isShouldFailOnError()) {
+            if (shouldFailOnError()) {
                 throw Fail.withInternalError(ex);
             }
             logSuppressed(ex);
@@ -74,7 +75,11 @@ final class ErrorHandler {
         return Optional.empty();
     }
 
-    private boolean isShouldFailOnError() {
+    public static boolean shouldFailOnError(final Settings settings) {
+        return settings.get(Keys.FAIL_ON_ERROR) || SystemProperties.isFailOnErrorEnabled();
+    }
+
+    private boolean shouldFailOnError() {
         return isFailOnErrorSettingEnabled || SystemProperties.isFailOnErrorEnabled();
     }
 

--- a/instancio-core/src/main/java/org/instancio/internal/generator/util/CollectionGenerator.java
+++ b/instancio-core/src/main/java/org/instancio/internal/generator/util/CollectionGenerator.java
@@ -22,9 +22,11 @@ import org.instancio.generator.Hints;
 import org.instancio.generator.hints.CollectionHint;
 import org.instancio.generator.specs.CollectionGeneratorSpec;
 import org.instancio.internal.ApiValidator;
+import org.instancio.internal.ErrorHandler;
 import org.instancio.internal.generator.AbstractGenerator;
 import org.instancio.internal.generator.InternalGeneratorHint;
 import org.instancio.internal.util.CollectionUtils;
+import org.instancio.internal.util.ExceptionUtils;
 import org.instancio.internal.util.Fail;
 import org.instancio.internal.util.NumberUtils;
 import org.instancio.internal.util.Sonar;
@@ -127,7 +129,13 @@ public class CollectionGenerator<T> extends AbstractGenerator<Collection<T>> imp
         try {
             return (Collection<T>) collectionType.getDeclaredConstructor().newInstance();
         } catch (Exception ex) {
-            throw Fail.withFataInternalError("Error creating instance of: %s", collectionType, ex);
+            final String msg = String.format("Error creating instance of: %s", collectionType);
+
+            if (ErrorHandler.shouldFailOnError(getContext().getSettings())) {
+                throw Fail.withFataInternalError(msg, ex);
+            }
+            ExceptionUtils.logException(msg, ex);
+            return null; //NOPMD
         }
     }
 

--- a/instancio-core/src/main/java/org/instancio/internal/generator/util/MapGenerator.java
+++ b/instancio-core/src/main/java/org/instancio/internal/generator/util/MapGenerator.java
@@ -22,8 +22,10 @@ import org.instancio.generator.Hints;
 import org.instancio.generator.hints.MapHint;
 import org.instancio.generator.specs.MapGeneratorSpec;
 import org.instancio.internal.ApiValidator;
+import org.instancio.internal.ErrorHandler;
 import org.instancio.internal.generator.AbstractGenerator;
 import org.instancio.internal.generator.InternalGeneratorHint;
+import org.instancio.internal.util.ExceptionUtils;
 import org.instancio.internal.util.Fail;
 import org.instancio.internal.util.NumberUtils;
 import org.instancio.internal.util.Sonar;
@@ -138,7 +140,13 @@ public class MapGenerator<K, V> extends AbstractGenerator<Map<K, V>> implements 
         try {
             return (Map<K, V>) mapType.getDeclaredConstructor().newInstance();
         } catch (Exception ex) {
-            throw Fail.withFataInternalError("Error creating instance of: %s", mapType, ex);
+            final String msg = String.format("Error creating instance of: %s", mapType);
+
+            if (ErrorHandler.shouldFailOnError(getContext().getSettings())) {
+                throw Fail.withFataInternalError(msg, ex);
+            }
+            ExceptionUtils.logException(msg, ex);
+            return null; //NOPMD
         }
     }
 

--- a/instancio-tests/instancio-core-tests/src/test/java/org/instancio/internal/generator/util/CollectionGeneratorTest.java
+++ b/instancio-tests/instancio-core-tests/src/test/java/org/instancio/internal/generator/util/CollectionGeneratorTest.java
@@ -95,7 +95,16 @@ class CollectionGeneratorTest {
     }
 
     @Test
-    void shouldFailOnInstantiationError() {
+    void shouldNotFailOnInstantiationErrorByDefault() {
+        final CollectionGenerator<?> generator = generator().subtype(List.class);
+
+        assertThat(generator.generate(random)).isNull();
+    }
+
+    @Test
+    void shouldFailOnInstantiationErrorWhenFailOnErrorIsEnabled() {
+        settings.set(Keys.FAIL_ON_ERROR, true);
+
         final CollectionGenerator<?> generator = generator().subtype(List.class);
 
         assertThatThrownBy(() -> generator.generate(random))

--- a/instancio-tests/instancio-core-tests/src/test/java/org/instancio/internal/generator/util/MapGeneratorTest.java
+++ b/instancio-tests/instancio-core-tests/src/test/java/org/instancio/internal/generator/util/MapGeneratorTest.java
@@ -96,7 +96,16 @@ class MapGeneratorTest {
     }
 
     @Test
-    void shouldFailOnInstantiationError() {
+    void shouldNotFailOnInstantiationErrorByDefault() {
+        final MapGenerator<?, ?> generator = generator().subtype(Map.class);
+
+        assertThat(generator.generate(random)).isNull();
+    }
+
+    @Test
+    void shouldFailOnInstantiationErrorWhenFailOnErrorIsEnabled() {
+        settings.set(Keys.FAIL_ON_ERROR, true);
+
         final MapGenerator<?, ?> generator = generator().subtype(Map.class);
 
         assertThatThrownBy(() -> generator.generate(random))

